File: code/library/src/main/java/org/fbme/lib/iec61499/parser/SegmentTypeConverter.java
package org.fbme.lib.iec61499.parser;


import org.fbme.lib.common.Identifier;
import org.fbme.lib.iec61499.declarations.SegmentTypeDeclaration;
import org.jetbrains.annotations.Nullable;

public class SegmentTypeConverter extends DeclarationConverterBase<SegmentTypeDeclaration> {

    public SegmentTypeConverter(ConverterArguments arguments) {
        super(arguments);
    }

    @Override
    protected SegmentTypeDeclaration extractDeclarationBody(@Nullable Identifier identifier) {
        SegmentTypeDeclaration std = myFactory.createSegmentTypeDeclaration(identifier);
        ParameterDeclarationConverter.extractAll(this, std.getParameters());
        return std;
    }
}

File: code/library/src/main/kotlin/org/fbme/lib/iec61499/parser/SegmentTypeConverter.kt
package org.fbme.lib.iec61499.parser

import org.fbme.lib.common.Identifier
import org.fbme.lib.iec61499.declarations.SegmentTypeDeclaration

class SegmentTypeConverter(arguments: ConverterArguments) :
    DeclarationConverterBase<SegmentTypeDeclaration>(arguments) {
    override fun extractDeclarationBody(identifier: Identifier?): SegmentTypeDeclaration {
        val std = factory.createSegmentTypeDeclaration(identifier)
        ParameterDeclarationConverter.extractAll(this, std.parameters)
        return std
    }
}
